subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/resolver/KotlinBuildScriptDependenciesResolver.kt

```diff
@@ -221,3 +227,43 @@ object RequestQueue {
         }
     }
 }
+
+
+private
+fun Report.exceptions(scriptFile: File?, exceptions: List<Exception>) =
+    exceptions.mapNotNull { inferLineNumberAndMessageOrNull(scriptFile, it) }.forEach { (lineNumber, message) ->
+        error(message, Position(lineNumber - 1, 1))
+    }
+
+
+private
+tailrec fun inferLineNumberAndMessageOrNull(scriptFile: File?, ex: Throwable): Pair<Int, String>? {
+    if (ex.isReportedLocationAwareException && ex.appliesTo(scriptFile)) {
+        ex.locationAwareLineNumber?.let { lineNumber ->
+            return Pair(lineNumber, ex.locationAwareMessage)
+        }
+    }
+    val cause = ex.cause ?: return null
+    return inferLineNumberAndMessageOrNull(scriptFile, cause)
+}
+
+
+private
+val Throwable.isReportedLocationAwareException
+    get() = this::class.java.name == LocationAwareException::class.java.name
```

Comment:
❓ Why comparing the type name instead of checking the type via `is` or `as?`?

Reply:
Reflection is used on the location aware exceptions because their java class comes from another class loader. Using `is` or `as?` doesn't match.